### PR TITLE
[5.9] Editorial review changes of PackageDescription documentation.

### PR DESCRIPTION
**Pull request by zkiraly**
Editorial review changes for the `PackageDescription` documentation update that was applied in 942a1e8

### Motivation:

To assure the high quality of the `PackageDescription` documentation, an editorial review was conducted offline. The recommended changes are now presented for the community to ensure accuracy.

### Modifications:

Only DocC documentation was changed. Several files in `PackageDescription` were changed, and the bulk of the changes are in `Target.swift`.

### Result:

After the PR is merged, the `PackageDescription` documentation will be updated with the editorial review results.

### CODE REVIEW

**Review by tomerd — APPROVED** [review]

File: Sources/PackageDescription/Target.swift [hunk 1]
```diff
@@ -1462,7 +1462,7 @@ public extension PluginCommandIntent {
         return documentationGeneration
     }
 
-    /// The plugin formats source code.
+    /// The plug-in format's source code.
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:1465]: This looks wrong, we mean the verb `format` here. A `sourceCodeFormatting` plugin can be used to format the user's code.

**Review by neonichu — CHANGES_REQUESTED** [review]

**Comment by zkiraly** [comment]
@swift-ci please smoke test.

File: Sources/PackageDescription/Target.swift (hunk 1 above)
Comment (zkiraly) [reply Sources/PackageDescription/Target.swift]: Thanks Boris, I applied your change.

**Comment by neonichu** [comment]
CI failure looks infra related

```
Cloning into 'swift-atomics'...
fatal: unable to access 'https://github.com/apple/swift-atomics.git/': Received HTTP code 503 from proxy after CONNECT
```

**Review by neonichu — APPROVED** [review]

**Comment by zkiraly** [comment]
@swift-ci please smoke test.

**Comment by zkiraly** [comment]
@swift-ci please smoke test.

**Pull request merged**